### PR TITLE
git read issues

### PR DESCRIPTION
**Pull request by piotrek6641**

### CODE REVIEW

**Review by patuwwy — CHANGES_REQUESTED** [review]
please use linter

File: js/git-read-issues/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+> ! All issues that has been read by this sequence will be labeled as read in a github
+
+## Setup
+
+Change data in `ghdata.json` to one that applies to you
+
+```json
+{
+    "owner": "owner", //Repository owner
+    "auth": "KEY", //API-KEY
```
Comment (alicja-gruzdz) [inline js/git-read-issues/README.md:16]: like in the Clickup Sequence, can we please move the key to be passed as an argument of the Sequence? 
thanks :)

**Review by alicja-gruzdz — CHANGES_REQUESTED** [review]

File: js/git-read-issues/README.md (hunk 1 above)
Comment (piotrek6641) [reply js/git-read-issues/README.md]: Thanks, this was already implemented. I've forgot to change readme file

File: js/git-read-issues/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Git read issues
+
+This sequence reads all of the github issues for a given repositories
+
+it prepares the data in a format that will be used in [clickup-add-issues]() and send it to a topic.
```
Comment (patuwwy) [inline js/git-read-issues/README.md:5]:
```suggestion
It prepares the data in a format that will be used in [clickup-add-issues]() and send it to a topic.
```

File: js/git-read-issues/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        "owner":"owner",
+        "repo":"example"
+    },
+    {
+        "owner":"owner",
+        "repo":"example1"
+    }
+]
+}
```
Comment (patuwwy) [inline js/git-read-issues/README.md:25]:
```suggestion
{
    "repos": [
        {
            "owner": "owner",
            "repo": "example"
        },
        {
            "owner":"owner",
            "repo":"example1"
        }
    ]
}
```

File: js/git-read-issues/ghdata.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    {
+        "owner":"owner",
+        "repo":"example"
+    },
+    {
+        "owner":"owner",
+        "repo":"example1"
+    }
+]
+}
```
Comment (patuwwy) [inline js/git-read-issues/ghdata.json:12]:
```suggestion
{
    "repos": [
        {
            "owner": "owner",
            "repo": "example"
        },
        {
            "owner": "owner",
            "repo": "example1"
        }
    ]
}
```

File: js/git-read-issues/src/github.ts
Comment (patuwwy) [inline js/git-read-issues/src/github.ts:1]: pls fix formatting

File: js/git-read-issues/src/index.ts
Comment (patuwwy) [inline js/git-read-issues/src/index.ts:1]: formatting, empty lines missing

File: js/git-read-issues/src/github.ts
Comment (piotrek6641) [reply js/git-read-issues/src/github.ts:1]: fixed with commit [5c8f7d7](https://github.com/scramjetorg/reference-apps/pull/20/commits/5c8f7d747900eda274aa4fb794431aa5c233a87a)

File: js/git-read-issues/src/githubClient.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                owner: this.owner,
+                repo: this.repo,
+                issue_number: e.number,
+                labels: [readLabel]
+            });
+        });
+
+        return gitRequestResponse.data;
+    }
+    async gitHubFilter() {
```
Comment (patuwwy) [inline js/git-read-issues/src/githubClient.ts:42]:
```suggestion

    async gitHubFilter() {
```

File: js/git-read-issues/src/githubClient.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        await this.gitHubFilter().then((result) =>
+            result.map((e) => {
+                if (typeof e.body === "string") {
+                    const entry = new Issue(e.number, e.title, e.body, this.repo, e.labels, this.owner);
+
+                    issuesArr.push(entry);
+                }
+            })
+        );
+        return issuesArr;
```
Comment (patuwwy) [inline js/git-read-issues/src/githubClient.ts:63]:
```suggestion

        return issuesArr;
```

File: js/git-read-issues/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+import { ReadableApp } from "@scramjet/types";
+import * as ghSettings from "../ghdata.json";
+import { PassThrough } from "stream";
+
+function labelHelper(labels: any[], repo: string): Array<string> {
+    return labels
+        .reduce((acc, c) => c !== "read" && c.name !== "read" ? acc.concat([c.name]) : acc, [])
+        .concat([repo]);
+}
+const output: PassThrough & { topic: string; contentType: string } = Object.assign(
```
Comment (patuwwy) [inline js/git-read-issues/src/index.ts:11]:
```suggestion

const output: PassThrough & { topic: string; contentType: string } = Object.assign(
```

File: js/git-read-issues/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                        name: issue.title,
+                        description: issue.body,
+                        tags: labelHelper(issue.labels, issue.repo)
+                    }) + "\n"
+                );
+            }
+        })
+    );
+}
+const app: ReadableApp<any> = async function (_stream, interval: number = 1000 * 30, apiKey: string) {
```
Comment (patuwwy) [inline js/git-read-issues/src/index.ts:34]:
```suggestion

const app: ReadableApp<any> = async function (_stream, interval: number = 1000 * 30, apiKey: string) {
```

File: js/git-read-issues/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    await main(apiKey);
+    setInterval(async () => {
```
Comment (patuwwy) [inline js/git-read-issues/src/index.ts:36]:
```suggestion

    setInterval(async () => {
```

File: js/git-read-issues/src/githubClient.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
+import { Octokit } from "octokit";
+import { ListUserReposResponse } from "./types/types";
+import { Issue } from "./issue";
+const readLabel = "read";
```
Comment (patuwwy) [inline js/git-read-issues/src/githubClient.ts:4]:
```suggestion

const readLabel = "read";
```

**Review by patuwwy — APPROVED** [review]

**Pull request merged**